src/messaging/messaging-api-request.ts

```diff
@@ -96,76 +94,70 @@ export class FirebaseMessagingRequestHandler {
    * @return {Promise<object>} A promise that resolves with the response.
    */
   public invokeRequestHandler(host: string, path: string, requestData: object): Promise<object> {
-    return this.signedApiRequestHandler.sendRequest(
-      host,
-      FIREBASE_MESSAGING_PORT,
-      path,
-      FIREBASE_MESSAGING_HTTP_METHOD,
-      requestData,
-      FIREBASE_MESSAGING_HEADERS,
-      FIREBASE_MESSAGING_TIMEOUT,
-    ).then((response) => {
+    const request: HttpRequestConfig = {
+      method: FIREBASE_MESSAGING_HTTP_METHOD,
+      url: `https://${host}${path}`,
+      data: requestData,
+      headers: FIREBASE_MESSAGING_HEADERS,
+      timeout: FIREBASE_MESSAGING_TIMEOUT,
+    };
+    return this.httpClient.send(request).then((response) => {
       // Send non-JSON responses to the catch() below where they will be treated as errors.
-      if (typeof response === 'string') {
-        return Promise.reject({
-          error: response,
-          statusCode: 200,
-        });
+      if (!response.isJson) {
```

Comment:
I think it would be better to create a property or method to determine whether the response is valid, something like `isValid()` or `valid`.

Reply:
Renamed to `contentJson` and added some documentation. Didn't feel like naming this just `json`, since that seems to imply this returns a json object. wdyt?